src/ui/projects/CreateProjectPage.js

```diff
@@ -0,0 +1,164 @@
+        }
+
+        search.set(name, nextParams[name]);
+      }
+
+      history.push(`/projects/create?${search}`);
+
+      setParams(nextParams);
+    },
+    [history]
+  );
+
+  const onChangeQuery = useCallback(
+    value => {
+      updateParams({
+        source: "scene_listings",
+        filter: "remixable",
+        q: value
+      });
+    },
+    [updateParams]
+  );
+
+  const onSetFeaturedRemixable = useCallback(() => {
+    updateParams({
+      ...params,
+      filter: "featured-remixable",
+      q: ""
+    });
+  }, [updateParams, params]);
+
+  const onSetAll = useCallback(() => {
+    updateParams({
+      ...params,
+      filter: "remixable",
+      q: ""
+    });
+  }, [updateParams, params]);
+
+  const onSelectScene = useCallback(
+    scene => {
+      const search = new URLSearchParams();
+      search.set("sceneId", scene.id);
+      history.push(`/projects/new?${search}`);
+    },
+    [history]
+  );
+
+  const { loading, error, entries, hasMore, loadMore } = usePaginatedSearch(
+    `${api.apiURL}/api/v1/media/search`,
+    params
+  );
+
+  const filteredEntries = entries.map(result => ({
+    ...result,
+    url: `/projects/new?sceneId=${result.id}`,
+    thumbnail_url: result && result.images && result.images.preview && result.images.preview.url
+  }));
+
+  return (
+    <>
+      <NavBar />
+      <main>
+        <ProjectsSection>
+          <ProjectsContainer>
+            <ProjectsHeader>
+              <h1>New Project</h1>
+              <PrimaryLink to="/projects">Back to projects</PrimaryLink>
+            </ProjectsHeader>
+            <ProjectGridContainer>
+              <ProjectGridHeader>
+                <ProjectGridHeaderRow>
+                  <Filter onClick={onSetFeaturedRemixable} active={params.filter === "featured-remixable"}>
+                    Featured
+                  </Filter>
+                  <Filter onClick={onSetAll} active={params.filter === "remixable"}>
+                    All
+                  </Filter>
+                  <Separator />
+                  <SearchInput placeholder="Search scenes..." value={params.q} onChange={onChangeQuery} />
+                </ProjectGridHeaderRow>
+                <ProjectGridHeaderRow>
+                  <Button as={Link} to="/projects/new">
+                    New Empty Project
+                  </Button>
+                </ProjectGridHeaderRow>
+              </ProjectGridHeader>
+              <ProjectGridContent>
+                <ScrollToTop />
+                {error && <ErrorMessage>{error.message}</ErrorMessage>}
+                {!error && (
+                  <InfiniteScroll
+                    initialLoad={false}
+                    pageStart={0}
+                    loadMore={loadMore}
+                    hasMore={hasMore}
+                    threshold={100}
+                    useWindow={true}
+                  >
+                    <ProjectGrid
+                      projects={filteredEntries}
```

Comment:
Bit weird that these are actually scene listings and not projects. Kind of surprising that it happens to work out. Would be pretty easy to mess up in the future though if their shapes diverge more.

Reply:
Yeah, agreed, see above. I'll refactor with the next type of media to use this grid component.